src/frontend/js/components/AshleyEditor/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { convertFromRaw, convertToRaw, EditorState, RichUtils } from 'draft-js';
+import createLinkPlugin from 'draft-js-anchor-plugin';
+import {
+  BoldButton,
+  HeadlineOneButton,
+  HeadlineThreeButton,
+  HeadlineTwoButton,
+  ItalicButton,
+  OrderedListButton,
+  UnderlineButton,
+  UnorderedListButton,
+} from 'draft-js-buttons';
+import createEmojiPlugin, {
+  EmojiPlugin,
+  EmojiPluginConfig,
+} from 'draft-js-emoji-plugin';
+import Editor from 'draft-js-plugins-editor';
+import PluginEditor from 'draft-js-plugins-editor/lib';
+import createToolbarPlugin, { Separator } from 'draft-js-static-toolbar-plugin';
+import React, { ComponentType, Ref } from 'react';
+import ReactDOM from 'react-dom';
+
+interface MyEditorProps {
+  autofocus?: boolean;
+  placeholder?: string;
+  target: HTMLInputElement;
+  emojiConfig?: EmojiPluginConfig;
+}
+
+// Initialize the Toolbar plugin
+const toolbarPlugin = createToolbarPlugin();
+const { Toolbar } = toolbarPlugin;
+
+// Initialize the link plugin
+const linkPlugin = createLinkPlugin({
+  linkTarget: '_blank',
+});
+const { LinkButton } = linkPlugin;
+
+export class AshleyEditor extends React.Component<MyEditorProps, any> {
```

Comment:
All our react component are written using hooks and functions. Can you do the same here ?

Reply:
I tried (very hard), but I have an issue with some draft-js plugins that are not well initialized after the rewrite :cry:

Reply:
Don't worry we will work on this in another PR